From c88aa62894ef2d0bdedbdefe870adc528533bd3e Mon Sep 17 00:00:00 2001
From: Mike Looijmans <mike.looijmans@topic.nl>
Date: Mon, 30 Sep 2019 14:44:23 +0200
Subject: [PATCH] zynqmp-topic-miamimp: fixup devicetree for 4.19 kernel

Rename "clk" to "zynqmp_clk"
Remove fpga_full (already in zynqmp.dtsi)
Fix dtc warnings
Enable pcap
Move PL components to overlay
---
 .../zynqmp-topic-miamimp-florida-gen.dts      | 244 +-----------------
 .../boot/dts/xilinx/zynqmp-topic-miamimp.dts  |  33 +--
 2 files changed, 16 insertions(+), 261 deletions(-)

diff --git a/arch/arm64/boot/dts/xilinx/zynqmp-topic-miamimp-florida-gen.dts b/arch/arm64/boot/dts/xilinx/zynqmp-topic-miamimp-florida-gen.dts
index e67323a15b2b..347f1d9b8acd 100644
--- a/arch/arm64/boot/dts/xilinx/zynqmp-topic-miamimp-florida-gen.dts
+++ b/arch/arm64/boot/dts/xilinx/zynqmp-topic-miamimp-florida-gen.dts
@@ -41,25 +41,13 @@
 			linux,code = <116>; /* KEY_POWER */
 			gpio-key,wakeup;
 		};
-		/* On-board push buttons in reference design */
-		s2 {
-			label = "s2";
-			gpios = <&gpio 95 1>; /* EMIO 17, active low */
-			linux,code = <60>; /* KEY_F2 */
-			gpio-key,wakeup;
-		};
-		s3 {
-			label = "s3";
-			gpios = <&gpio 96 1>; /* EMIO 18, active low */
-			linux,code = <61>; /* KEY_F3 */
-			gpio-key,wakeup;
-		};
 	};
 
 	gpio-poweroff {
 		compatible = "gpio-poweroff";
 		gpios = <&gpioexf 5 1>; /* Output 5, active low */
 		input;
+		force-pm-power-off; /* Replace the PSCI power-off callback */
 	};
 
 	charger: battery-charger {
@@ -97,40 +85,18 @@
 /* SD card with level shifter */
 &sdhci1 {
 	status = "okay";
-	no-1-8-v;	/* for 1.0 silicon */
 	xlnx,mio_bank = <1>;
+	no-1-8-v; /* No high-speed capability */
 	disable-wp; /* We don't have a write-protect detection */
 };
 
-&serdes {
-	status = "okay";
-};
-
 &uart1 {
 	status = "okay";
 };
 
-&sata {
-	status = "okay";
-	/* SATA OOB timing settings */
-	ceva,p0-cominit-params = /bits/ 8 <0x18 0x40 0x18 0x28>;
-	ceva,p0-comwake-params = /bits/ 8 <0x06 0x14 0x08 0x0E>;
-	ceva,p0-burst-params = /bits/ 8 <0x13 0x08 0x4A 0x06>;
-	ceva,p0-retry-params = /bits/ 16 <0x96A4 0x3FFC>;
-	ceva,p1-cominit-params = /bits/ 8 <0x18 0x40 0x18 0x28>;
-	ceva,p1-comwake-params = /bits/ 8 <0x06 0x14 0x08 0x0E>;
-	ceva,p1-burst-params = /bits/ 8 <0x13 0x08 0x4A 0x06>;
-	ceva,p1-retry-params = /bits/ 16 <0x96A4 0x3FFC>;
-	/* For now, only try one instance */
-	phy-names = "sata-phy", "sata-phy";
-	/* Use refclock 0 at 150MHz for both instances */
-	phys = <&lane0 PHY_TYPE_SATA 0 0 150000000>,
-	       <&lane1 PHY_TYPE_SATA 1 0 150000000>;
-	/* Using SI514 on miami through logic as REFCLK0. Must be programmed to 150MHz. */
-	assigned-clocks = <&si514s>;
+/* Setup for SATA serdes clock */
+&si514s {
 	assigned-clock-rates = <150000000>;
-	/* Apparently doesn't work this way? */
-	clocks = <&si514s>;
 };
 
 /* I2C bus layout */
@@ -160,33 +126,6 @@
 				 *};
 				 */
 			};
-			/* HDMI OUT */
-			adv7511: adv7511@39 {
-				compatible = "adi,adv7511";
-				reg = <0x39>, <0x3f>;
-				reg-names = "primary", "edid";
-				adi,input-depth = <8>;
-				adi,input-colorspace = "rgb";
-				adi,input-clock = "1x";
-				adi,input-style = <1>;
-				adi,input-justification = "right";
-				adi,clock-delay = <0>;
-				#sound-dai-cells = <0>;
-
-				ports {
-					#address-cells = <1>;
-					#size-cells = <0>;
-					port@0 {
-						reg = <0>;
-						adv7511_in: endpoint {
-							remote-endpoint = <&axi_hdmi_out>;
-						};
-					};
-					port@1 {
-						reg = <1>;
-					};
-				};
-			};
 		};
 		i2cmux_2: i2c@2 {
 			/* GEN */
@@ -235,7 +174,7 @@
 				compatible = "silabs,si514";
 			};
 			/* Battery gauge */
-			fuelgauge: battery-gauge {
+			fuelgauge: battery-gauge@64 {
 				compatible = "ltc2943";
 				reg = <0x64>;
 				lltc,resistor-sense = <15>; /* 15 mOhm */
@@ -243,59 +182,6 @@
 				power-supplies = <&charger>;
 			};
 		};
-		i2c@3 {
-			/* HDMI IN */
-			#address-cells = <1>;
-			#size-cells = <0>;
-			reg = <3>;
-			/* copied from zynq-zed-imageon.dts */
-			adv7611@4c {
-				compatible = "adi,adv7611";
-				reg = <0x4c>;
-				#address-cells = <0x1>;
-				#size-cells = <0x0>;
-				bus-width = <24>;
-				hsync-active = <0>; /* Parallel bus */
-				vsync-active = <0>;
-				interrupt-parent = <&gpio>;
-				interrupts = <99 4>; /* EMIO 21 */
-				adi,int1-config = <1>; /* ADV76XX_INT1_CONFIG_ACTIVE_LOW */
-				reset-gpios = <&gpio 91 0>; /* EMIO 13 */
-				port@0 {
-					reg = <0x0>;
-				};
-				port@1 {
-					reg = <0x1>;
-					adv7611: endpoint {
-						bus-width = <16>;
-						remote-endpoint = <&axi_hdmi_rx>;
-					};
-				};
-			};
-		};
-	};
-};
-
-/* Ethernet: Micrel PHY via logic to RGMII, on MDIO#3 */
-&gem3 {
-	status = "okay";
-	phy-handle = <&gmiitorgmii>;
-	phy-mode = "gmii";
-	/* gem3 is the eth0 controller, hence we'll use the "0" mac for that */
-	nvmem-cells = <&gem0_mac>;
-	nvmem-cell-names = "mac-address";
-	phy3: phy@3 {
-		device_type = "ethernet-phy";
-		reg = <3>;
-		interrupt-parent = <&gpio>;
-		interrupts = <97 0x8>; /* GPIO 78+16+3, IRQ_TYPE_LEVEL_LOW */
-		rxc-skew-ps = <1080>; /* Skew RX clock pad by 180 ps */
-		txc-skew-ps = <840>;  /* Skew TX clock pad by -60 ps */
-	};
-	gmiitorgmii: gmiitorgmii@8 {
-		compatible = "xlnx,gmii-to-rgmii-1.0";
-		reg = <8>;
-		phy-handle = <&phy3>;
 	};
 };
 
@@ -310,9 +196,10 @@
 	/* see: https://forums.xilinx.com/t5/Embedded-Linux/Zynqmp-USB2-0/td-p/790522/page/2 */
 	/delete-property/ clocks;
 	/delete-property/ clock-names;
-	clocks = <&clk 32>;
+	clocks = <&zynqmp_clk USB0_BUS_REF>;
 	clock-names = "bus_clk";
 };
+
 &dwc3_0 {
 	status = "okay";
 	dr_mode = "host"; /* "host", "peripheral" or "otg" */
@@ -324,123 +211,6 @@
 	/delete-property/ phys;
 };
 
-/* FPGA content (reference design) */
-&amba {
-	axi_vdma_0: axivdma@a0040000 {
-		compatible = "xlnx,axi-vdma-1.00.a";
-		reg = <0x0 0xa0040000 0x0 0x10000>;
-		#address-cells = <1>;
-		#size-cells = <1>;
-		#dma-cells = <1>;
-		#dma-channels = <1>;
-		clock-names = "s_axi_lite_aclk", "m_axi_mm2s_aclk", "m_axi_mm2s_aclk";
-		clocks = <&clk 71>, <&clk 72>, <&clk 72>; /* fclk0 and fclk1 */
-		interrupt-parent = <&gic>;
-		interrupts = <0 89 4>;
-		dma-ranges = <0x00000000 0x00000000 0x80000000>;
-		xlnx,num-fstores = <3>;
-		xlnx,flush-fsync = <1>;
-		xlnx,addrwidth = <32>;
-		dma-channel@a0040000 {
-			compatible = "xlnx,axi-vdma-mm2s-channel";
-			interrupts = <0 89 4>; /* pl_ps_0[0] */
-			xlnx,datawidth = <64>;
-			xlnx,device-id = <0x0>;
-		};
-	};
-
-	axi_hdmi_0: axi_hdmi@a0030000 {
-		compatible = "adi,axi-hdmi-tx-1.00.a";
-		reg = <0x0 0xa0030000 0x0 0x10000>;
-		adi,is-rgb;
-		dmas = <&axi_vdma_0 0>;
-		dma-names = "video";
-		clocks = <&clockgen 3>;
-		port {
-			axi_hdmi_out: endpoint {
-				remote-endpoint = <&adv7511_in>;
-			};
-		};
-	};
-
-	axi_spdif_tx_0: axi-spdif-tx@a00f0000 {
-		status = "disabled"; /* DMA connection unknown */
-		compatible = "adi,axi-spdif-tx-1.00.a";
-		reg = <0x0 0xa00f0000 0x0 0x1000>;
-		/* TODO:  dmas = <&dmac_s 0>; */
-		dma-names = "tx";
-		clocks = <&clk 71>, <&clockgen 1>, <&clockgen 1>;
-		clock-names = "axi", "ref", "spdif";
-
-		#sound-dai-cells = <0>;
-	};
-
-
-	axi_hdmi_rx_core: axi-hdmi-rx@a0020000 {
-		compatible = "adi,axi-hdmi-rx-1.00.a";
-		reg = <0x0 0xa0020000 0x0 0x10000>;
-		dmas = <&axi_dmac 0>;
-		dma-names = "rx";
-		gpios = <&gpioexf 3 0>; /* HPD, hotplug control (?) */
-
-		port {
-			#address-cells = <0>;
-			#size-cells = <0>;
-
-			axi_hdmi_rx: endpoint {
-				remote-endpoint = <&adv7611>;
-				bus-width = <24>;
-			};
-		};
-	};
-
-	axi_dmac: dmac@a0000000 {
-		compatible = "adi,axi-dmac-1.00.a";
-		reg = <0x0 0xa0000000 0x0 0x10000>;
-		#dma-cells = <1>;
-		interrupt-parent = <&gic>;
-		interrupts = <0 90 4>; /* pl_ps_0[1] */
-		clocks = <&clk 71>; /* fclk0 */
-
-		adi,channels {
-			#size-cells = <0>;
-			#address-cells = <1>;
-			dma-channel@0 {
-				reg = <0>;
-				adi,length-width = <14>;
-				adi,source-bus-width = <64>;
-				adi,source-bus-type = <2>;
-				adi,destination-bus-width = <64>;
-				adi,destination-bus-type = <0>;
-				adi,2d;
-			};
-		};
-	};
-};
-
-/ {
-	adv7511_hdmi_snd {
-		compatible = "simple-audio-card";
-		simple-audio-card,name = "HDMI monitor";
-		simple-audio-card,widgets =
-			"Speaker", "Speaker";
-		simple-audio-card,routing =
-			"Speaker", "TX";
-
-		simple-audio-card,dai-link@0 {
-			format = "spdif";
-			cpu {
-				sound-dai = <&axi_spdif_tx_0>;
-				frame-master;
-				bitclock-master;
-			};
-			codec {
-				sound-dai = <&adv7511>;
-			};
-		};
-	};
-};
-
 &pinctrl0 {
 	status = "okay";
 
diff --git a/arch/arm64/boot/dts/xilinx/zynqmp-topic-miamimp.dts b/arch/arm64/boot/dts/xilinx/zynqmp-topic-miamimp.dts
index 911a09d82751..456e0224a430 100644
--- a/arch/arm64/boot/dts/xilinx/zynqmp-topic-miamimp.dts
+++ b/arch/arm64/boot/dts/xilinx/zynqmp-topic-miamimp.dts
@@ -74,21 +74,10 @@
 		#phy-cells = <0>;
 		reset-gpios = <&gpioex 0 0>;
 	};
-
-	/* Full FPGA image */
-	fpga_full: fpga-full {
-		compatible = "fpga-region";
-		fpga-mgr = <&pcap>;
-		#address-cells = <1>;
-		#size-cells = <1>;
-		ranges;
-		firmware-name = "fpga.bin";
-	};
 };
 
 &pcap {
-	/* It's broken and just hangs the board */
-	status = "disabled";
+	status = "okay";
 };
 
 &gpio {
@@ -136,7 +125,7 @@
 		reg = <0x50>;
 		#address-cells = <1>;
 		#size-cells = <1>;
-		gem0_mac: mac@0x1e8 {
+		gem0_mac: mac@1e8 {
 			reg = <0x1e8 6>;
 		};
 		gem1_mac: mac@1ee {
@@ -181,21 +170,17 @@
 			label = "qspi-boot-bin";
 			reg = <0x00000 0x60000>;
 		};
-		partition@qspi-atf-spi-ub {
-			label = "qspi-atf-spi-ub";
-			reg = <0x60000 0x20000>;
-		};
-		partition@qspi-u-boot-bin {
-			label = "qspi-u-boot-bin";
-			reg = <0x80000 0xc0000>;
+		partition@qspi-u-boot-itb {
+			label = "qspi-u-boot-itb";
+			reg = <0x60000 0x100000>;
 		};
-		partition@qspi-u-boot-img {
-			label = "qspi-u-boot-img";
-			reg = <0x140000 0xc0000>;
+		partition@qspi-u-boot-env {
+			label = "qspi-u-boot-env";
+			reg = <0x160000 0x20000>;
 		};
 		partition@qspi-rootfs {
 			label = "qspi-rootfs";
-			reg = <0x200000 0x7e00000>;
+			reg = <0x180000 0x7e80000>;
 		};
 		/* Everything */
 		partition@qspi-all {
-- 
2.17.1

